cpp/src/datetime/datetime_util.cpp

```diff
@@ -43,11 +43,22 @@ std::pair<gdf_column, gdf_column> resolve_common_time_unit(gdf_column const& lhs
       rhs_unit = TIME_UNIT_ms;
     }
     if (lhs_unit != rhs_unit) {
-      gdf_dtype_extra_info out_info{std::max(lhs_unit, rhs_unit)};
-      if (lhs_unit > rhs_unit) {
-        rhs_out = cudf::cast(rhs, GDF_TIMESTAMP, out_info);
-      } else {
-        lhs_out = cudf::cast(lhs, GDF_TIMESTAMP, out_info);
+      if (lhs_unit == TIME_UNIT_ns) {
```

Comment:
I'd break this out into a standalone utility function as it may be useful elsewhere.

Something like:
```
gdf_time_unit common_resolution(gdf_time_unit lhs, gdf_time_unit rhs);
```

Where it returns the common resolution between the two specified in `lhs` and `rhs`.

Reply:
done!